docs/fornax-design/mizar_gateway_design

```diff
@@ -0,0 +1,54 @@
+# Adding Cross Cluster Gateways to Mizar
+
+A gateway is a field belonging to a mizar subnet object that all the traffics go to the gateway if there are any cross cluster requests. 
+
+```
+- name: Gateway
+  type: string
+  priority: 0
+  JSONPath: .spec.gateway
+  description: The IP of the gateway
+```
+
+Also, we add another field External to the subnet object that indicates wether the subnet belongs to the local cluster
+
+```
+- name: External
+  type: boolean
+  priority: 0
+  JSONPath: .spec.external
+  description: True means that the subnet does not belong to the local cluster; False means that the subnet belongs to the local cluster
+```
+
+## Control Plane
+For mizar control planes, we add the two fields above to subnet objects. The default value of gateway is empty and the one of external is false. If all the requests happen within the clusters, the two fields do not invovle at all.
+
+### Create a gateway droplet
+1. Deploy gateway endpoints
+2. Get gateway endpoint droplet ip 
+3. Add the gateway droplet ip to subnet
+4. The change triggers subnet workflows to delete the droplet and treat it as gateway droplet
+
+### Create an external subnet
+1. Deploy an external cluster
+2. Deploy a subnet and its gateway in the external cluster
+3. Create an external subnet with the gateway ip in a local cluster
+4. THe control plane does not trigger any other creation since the subnet is external
+
+### Ping external endpints in a local cluster
+1. Suppose the local subnet is 20.0.0.0/24 while the external subnet is 30.0.0.0/24. 
+2. A request is created in 20.0.0.10 to request the service at 30.0.0.10.
+3. The request will treat as external request because of the target ip 30.0.0.10.
+4. Given the source ip is 20.0.0.10, the transit xdp finds its local gateway and redirect the request to the gateway
+5. The gateway accepts its request and found its external subnet gateway
+6. The gateway redirects the request to the external gateaway
+
+## Data Plane
+A new type gateway droplet is added. Gateway bouncers will be deployed in that droplet and foward requests to the gateway
+
+### RPC service 
+1. A new service to update subnet by adding the gateway and the external fields
+2. A new service to update bouncers by assigning gateway ip
+
+### Transit XDP
+1. A bouncer metadata contain divider list
```

Comment:
why is bouncer involved in gateway?

Reply:
I am thinking to keep the origin bouncer and divider design without updating too many codes.  I am testing it to see if it works. I will roll back it to divider.